### PR TITLE
Add edit mode for site details 

### LINKED ISSUE #161: Site details form for editing sites
## Description

Route "edit site" button on cards to form input, and allow site metadata to be edited. We'll need to change the site details page to support a dynamic route `/site-details/[uuid]`, which will give us access to the corresponding (to the UUID) site's data to populate the form.

In addition, populate the context with current site details, specifically the `site_uuid`. Retain site details and populate form inputs/map with current details as defaults. 

## Todos

These points are rough guidelines. **Please** feel free to discuss the best way to design this component with others on the team!

- [x] Change route structure to support `site-details` for creating a new site and one for editing an existing site.
- [ ] Link existing `edit site` button on Site list page cards to this new route.
- [ ] Fetch site data and populate form fields with the data.
- [ ] Handle submission event, and instead of creating a new site, query the (not yet implemented) API endpoint to edit site data.

## Useful links

- [Next.js dynamic routes](https://nextjs.org/docs/routing/dynamic-routes)

## Time estimate

tbd

### PR DESCRIPTION
**Pull request by jackypark9852**
## Status:
:rocket: Ready

- Add separate routes for creating a new site and editing existing site information 
- Add auto fill functionality for existing site information  
## Description
This PR adds edit mode for the site details page. `/site-details/[uuid]` end point is for editing existing site. 

Fixes #161 

## Screenshots
image

### CODE REVIEW

File: pages/site-details/index.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import { withSites } from '../../lib/utils';
+import SiteDetails from '~/components/SiteDetails';
+
+enum Page {
+  Details = 'Details',
+  Location = 'Location',
+}
+
+const NewSiteDetails: FC = () => {
+  console.log('NewSiteDetails');
```
Comment (AndrewLester) [inline pages/site-details/index.tsx]:
```suggestion

```

File: pages/site-details/index.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { FC } from 'react';
+import { withSites } from '../../lib/utils';
+import SiteDetails from '~/components/SiteDetails';
+
+enum Page {
+  Details = 'Details',
+  Location = 'Location',
+}
+
```
Comment (AndrewLester) [inline pages/site-details/index.tsx]: Is this enum used?
```suggestion
```

File: pages/site-details/[uuid].tsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { FC } from 'react';
+import { withSites } from '../../lib/utils';
+import SiteDetails from '../../components/SiteDetails';
+import { useRouter } from 'next/router';
+
+const NewSiteDetails: FC = () => {
+  const router = useRouter();
+  const { uuid } = router.query;
+  console.log(uuid);
```
Comment (AndrewLester) [inline pages/site-details/[uuid].tsx]:
```suggestion

const NewSiteDetails: FC = () => {
  const router = useRouter();
  const { uuid } = router.query;
```

**Review by AndrewLester — APPROVED** [review]
This looks great, good work. Can you change the SiteCard "Edit Site" button's link to point to your new edit site details page?

**Review by ericcccsliu — APPROVED** [review]
LGTM

**Pull request merged**